packages/react-native-hid/android/src/main/java/com/ledgerwallet/hid/ReactHIDModule.java

```diff
@@ -29,30 +30,67 @@
 public class ReactHIDModule extends ReactContextBaseJavaModule {
 
     private final HashMap<String, HIDDevice> hidDevices = new HashMap<>();
+    private static final String ACTION_USB_ATTACHED  = "android.hardware.usb.action.USB_DEVICE_ATTACHED";
+    private static final String ACTION_USB_DETACHED  = "android.hardware.usb.action.USB_DEVICE_DETACHED";
+    private static final String ACTION_USB_PERMISSION = "com.ledgerwallet.hid.USB_PERMISSION";
+
+    private BroadcastReceiver receiver;
 
     public ReactHIDModule(ReactApplicationContext reactContext) {
         super(reactContext);
+        setDeviceConnectionReceiver();
     }
 
     @Override
     public String getName() {
         return "HID";
     }
 
+    private void setDeviceConnectionReceiver(){
+        IntentFilter filter = new IntentFilter();
+        filter.addAction(ACTION_USB_ATTACHED);
+        filter.addAction(ACTION_USB_DETACHED);
+
+        receiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context  context, Intent intent) {
+                switch (intent.getAction()) {
+                    case ACTION_USB_ATTACHED:
+
+                    UsbDevice device = (UsbDevice) intent.getExtras().get(UsbManager.EXTRA_DEVICE);
+                    getReactApplicationContext()
+                            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                            .emit("onDeviceConnect", buildMapFromDevice(device));
+                        break;
+                    case ACTION_USB_DETACHED:
+                        getReactApplicationContext()
+                                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                                .emit("onDeviceDisconnect", null);
```

Comment:
in this context we don't know *who* disconnects? we can't do the same buildMapFromDevice?

Reply:
Yes. but the id ends up being the same for two different devices on the same port. I plug my personal nano s and one of my dev nano s, it will not know which one is which. All this is because of a theoretical support for multiple dongles which will likely never happen

Reply:
For 1 device support this works..